<!DOCTYPE html>
<html>
  <head>
  <!--
    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.5.0/css/font-awesome.min.css">
   
    <link rel="stylesheet" href="/Users/kiverson/Downloads/font-awesome-4.5.0/css/font-awesome.min.css">
  --> 
    <link rel="stylesheet" href="font-awesome-4.5.0/css/font-awesome.min.css">
    <title>Microbial Community Ecology</title>
    <meta charset="utf-8">
    <style>
      @import url(https://fonts.googleapis.com/css?family=Droid+Serif);
      @import url(https://fonts.googleapis.com/css?family=Yanone+Kaffeesatz);
      @import url(https://fonts.googleapis.com/css?family=Ubuntu+Mono:400,700,400italic);
      body {
        font-family: 'Droid Serif';
      }
      h1, h2, h3 {
        font-family: 'Yanone Kaffeesatz';
        font-weight: 400;
        margin-bottom: 0;
      }
      .remark-slide-content h1 { font-size: 3em; }
      .remark-slide-content h2 { font-size: 2em; }
      .remark-slide-content h3 { font-size: 1.6em; }
      .footnote {
        position: absolute;
        bottom: 3em;
        left: 80px;
      }
      li p { line-height: 1.25em; }
      .red { color: #fa0000; }
      .large { font-size: 2em; }
      a, a > code {
        color: rgb(249, 38, 114);
        text-decoration: none;
      }
      code {
        background: #e7e8e2;
        border-radius: 5px;
      }
      .remark-code, .remark-inline-code { font-family: 'Ubuntu Mono'; }
      .remark-code-line-highlighted     { background-color: #373832; }
      .pull-left {
        float: left;
        width: 47%;
      }
      .pull-right {
        float: right;
        width: 47%;
      }
      .pull-right ~ p {
        clear: both;
      }
      #slideshow .slide .content code {
        font-size: 0.8em;
      }
      #slideshow .slide .content pre code {
        font-size: 0.9em;
        padding: 15px;
      }
      .inverse {
        background: #272822;
        color: #777872;
        text-shadow: 0 0 20px #333;
      }
      .inverse h1, .inverse h2 {
        color: #f3f3f3;
        line-height: 0.8em;
      }
      /* Slide-specific styling */
      #slide-inverse .footnote {
        bottom: 12px;
        left: 20px;
      }
      #slide-how .slides {
        font-size: 0.9em;
        position: absolute;
        top:  151px;
        right: 140px;
      }
      #slide-how .slides h3 {
        margin-top: 0.2em;
      }
      #slide-how .slides .first, #slide-how .slides .second {
        padding: 1px 20px;
        height: 90px;
        width: 120px;
        -moz-box-shadow: 0 0 10px #777;
        -webkit-box-shadow: 0 0 10px #777;
        box-shadow: 0 0 10px #777;
      }
      #slide-how .slides .first {
        background: #fff;
        position: absolute;
        top: 20%;
        left: 20%;
        z-index: 1;
      }
      #slide-how .slides .second {
        position: relative;
        background: #fff;
        z-index: 0;
      }
      /* Two-column layout */
      .left-column {
        color: #777;
        width: 20%;
        height: 92%;
        float: left;
      }
        .left-column h2:last-of-type, .left-column h3:last-child {
          color: #000;
        }
      .right-column {
        width: 75%;
        float: right;
        padding-top: 1em;
      }

      .first-column {
        width: 50%;
        float: left;
        padding-top: 1em;
      }
      .second-column {
        width: 50%;
        float: right;
        padding-top: 1em;
      }
    </style>
  </head>
  <body>
    <textarea id="source">


class: center, middle

# Microbial Community Ecology with Metagenomics
## Kathryn Iverson
## February 16th, 2016

.footnote[made with <i class="fa fa-heart" style="color:red"> </i> and [remark] ]

---

# Get in touch

<i class="fa fa-envelope"> kd.iverson@gmail.com </i> 

<i class="fa fa-twitter"> @kdiverson </i>  

<i class="fa fa-github-alt"> kdiverson </i> 

<i class="fa fa-file-code-o"></i> slides: kathryniverson.com/2Gtalk#1

---

# Agenda

1. About me

1. Projects

1. Analyzing metagenomes

1. Future directions

---

# Who I am

- BA in Biology from Earlham College

- MS in Bioinformatics from University of Michigan 

- Gut microbiome metagenomics projects

  - Colon cancer 

  - Clostridium difficile

  - IBD and Chron's microbiome

  - Healthy populations

        - Family project
        - Human Microbiome Project (HMP)

---
layout: false
# Projects
.left-column[
  ## Family data
]
.right-column[
- Follows a family of 8 (2 parents 6 children) over the course of a month

- 16S and metagenomic data

- Children represent a chronosequence

- Core microbiome via 16S

]

---
layout: false
.left-column[
  ## Family data
  ## Colon cancer
]
.right-column[
- Murine model of inflammation-associated colorectal cancer

  - Azoxymethane (AOM)/2% dextran sodium sulfate (DSS) model
  - AOM injection followed by 3 rounds of DSS

- Samples collected at day 0 and after each round of DSS

]

---
layout: false
.left-column[
  ## Family data
  ## Colon cancer
  ## HMP
]
.right-column[
  
  - Anterior nares
      - 87 metagenomes

- Buccal mucosa
      - 107 metagenomes

- Posterior fornix
      - 51 metagenomes

- Supragingival plaque
      - 118 metagenomes

- Stool
      - 138 metagenomes

- Tongue dorsum
      - 128 metagenomes

]

---
class: center, middle

# Processing the data

---
layout: false
.left-column[
  ## Quality trim reads
]
.right-column[
- SeqPrep to remove primers

- Sickle
]

---
.left-column[
  ## Quality trim reads
  ## Assemble
  ### why assemble?
]
.right-column[

A good analysis starts with a good assembly

- Reads are too short to contain full length genes

- Assembly reduces complexity of the dataset


Historically assemblers haven't been very good for metagenomes

- Assemblers assume even coverage

  - Metagenomes don't have even coverage

Work arounds

- Iterative assembly

  - Multiple velvet runs

]

---
.left-column[
  ## Quality trim reads
  ## Assemble
  ### why assemble?
  ### the right assembly
]
.right-column[

Solutions

- Megahit

    - Succinct de Bruijn graph assembler

    - Iterative k-mer assembly

    - CUDA
]


---
.left-column[
  ## Quality trim reads
  ## Assemble
  ## Call ORFs
]
.right-column[
Genes are the base of metagenomic analysis

- Provide insight into the genetic potential of the community

- 16S only part of the story

  - Tells you who's there, not what they're doing

]

---
.left-column[
  ## Quality trim reads
  ## Assemble
  ## Call ORFs
  ## Map reads
]
.right-column[
- Bowtie2

- Mapping reads provides an abundance count

- Normalize to gene length

```
ceiling(num_reads_mapped * read_length / len(gene))
```

]

---
.left-column[
  ## Quality trim reads
  ## Assemble
  ## Call ORFs
  ## Map reads
  ## BLAST
]
.right-column[
- All v all to get blast score ratio for clustering

- Similar to clustering 16S within a phylotype can cluster with a KEGG KO category

- Can also blast against specialized databases such at metaCyc, cazy, pfam

]


---
template: .inverse
class: center, middle

# OPFs

---
.left-column[
  ## OPFs
  ## What?
]
.right-column[

- Operational protein families are clusters of similar proteins

  - 30% sequence homology

- Creating OPFs is a database free method of grouping sequences 

- Analogous to 16S OTUs
]

---
.left-column[
  ## OPFs
  ## What?
  ## Why?
]
.right-column[

Databases are incomplete

- Genes present are limited to what we "know" or at least have seen before

- Most observed genes are still of unknown function

- Leaves a large portion of the data in a black box

  - Between 50%-80% of genes from gut datasets are unknown

- Provides a fine grain look at the data and can detect more subtle variation within and between communities
]

---
layout: false
.left-column[
  ## HMP data - buccal mucosa
]
.right-column[

![mucosa](mucosa.jpeg)

]

---
template: inverse
class: center, middle

# Analyzing the data

---
.left-column[
  ## Annotating OPFs
]
.right-column[

- Consensus annotation using existing databases 

  - Names
  - Functional domain

- Specialized databases are especially useful in this application

- OPFs are a unit analogous to OTUs

  - Diversity metrics

  - Machine learning applications

]


---
template: inverse
class: center, middle

# Machine learning

---
name: Machine learning

.left-column[
  ## What is it?
]
.right-column[
- Algorithms that evolve behaviors based on data

- Using examples of data to elucidate characteristics of that data (supervised learning)

- Find hidden structures in unlabeled data (unsupervised learning)

- Make generalizations about data

- Ability to accurately classify new data
]

---
.left-column[
  ## Random Forest
]
.right-column[

Algorithm based on decision trees

- Simply, a decision tree asks a series of questions about data in order to classify it

- Asks a question at each node and follows the path until it reaches the tip or leaf

- One child node per possible answer to each question

- This terminal node (leaf) is the classification

]

---
layout: false
.left-column[
  ## Feature selection
]
.right-column[
- Selecting a subset of data that has the relevant features for a learning algorithm

- Finding the characteristics of a dataset

- Also called gene selection in microarray analysis

- What OTUs are different between the groups?


]

---
layout: false
class: center, middle

## Colon cancer tumorigenesis


![AOM/DSS plot](aomdss.jpeg)


---
class: center, middle

# Putting it all together

---
layout: false
.left-column[
  ## Metadata
]
.right-column[

- Describes the environment in human friendly terms

]

---
layout: false
.left-column[
  ## Metadata
  ## 16S
]
.right-column[

- Tells us who's there

]

---
layout: false
.left-column[
  ## Metadata
  ## 16S
  ## Metagenomics
]
.right-column[

- Tells us what they can do 

]

---
layout: false
.left-column[
  ## Metadata
  ## 16S
  ## Metagenomics
  ## Transcriptomics
]
.right-column[

- Tells us what they're doing now

- Can model pathways

]

---
layout: false
.left-column[
  ## Metadata
  ## 16S
  ## Metagenomics
  ## Transcriptomics
  ## Metabolomics
]
.right-column[

- Tells us what compounds are available and/or being produced

]

---
class: center, middle

# Questions?


    </textarea>
<!--
    <script src="https://gnab.github.io/remark/downloads/remark-latest.min.js">

    <script src="/Users/kiverson/Downloads/remark-gh-pages/downloads/remark-0.13.min.js">
-->

   <script src="remark-0.13.min.js">
    </script>
    <script>
      var slideshow = remark.create();
    </script>
  </body>
</html>